jupyter_console/interactiveshell.py

```diff
@@ -597,3 +637,21 @@ def init_history(self):
         self.history_manager = ZMQHistoryManager(client=self.client)
         self.configurables.append(self.history_manager)
 
+    def get_source_and_reset(self):
```

Comment:
I think this is a fine DIY abstraction. Pre-mature optimization? Having written this kind of code before, I can think of reasons one might want to add other code in here, or overload it.